Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -1004,32 +1004,28 @@ void IfStatement::generate_bytecode(Bytecode::Generator& generator) const
     Bytecode::Op::Jump* true_block_jump { nullptr };
 
     generator.switch_to_basic_block(true_block);
+    generator.emit<Bytecode::Op::LoadImmediate>(js_undefined());
     m_consequent->generate_bytecode(generator);
     if (!generator.is_current_block_terminated())
         true_block_jump = &generator.emit<Bytecode::Op::Jump>();
 
     generator.switch_to_basic_block(false_block);
-    if (m_alternate) {
-        auto& end_block = generator.make_block();
+    auto& end_block = generator.make_block();
 
+    generator.emit<Bytecode::Op::LoadImmediate>(js_undefined());
+    if (m_alternate)
         m_alternate->generate_bytecode(generator);
-        if (!generator.is_current_block_terminated())
-            generator.emit<Bytecode::Op::Jump>().set_targets(
-                Bytecode::Label { end_block },
-                {});
+    if (!generator.is_current_block_terminated())
+        generator.emit<Bytecode::Op::Jump>().set_targets(
+            Bytecode::Label { end_block },
+            {});
```

Comment:
I know you just dedented this, but it should have curly braces as the body is more than one line (or just don't have the linebreaks tbh, it's not that long)

Reply:
Aye, I just blindly tapped `<<`. :) I've now changed it all to be on the same line.